integration/ballot.go

```diff
@@ -0,0 +1,465 @@
+	}
+
+	decryptBallots := types.CombineShares{
+		FormID: form.FormID,
+	}
+
+	data, err := decryptBallots.Serialize(serdecontext)
+	if err != nil {
+		return xerrors.Errorf("failed to serialize ballots: %v", err)
+	}
+
+	args := []txn.Arg{
+		{Key: native.ContractArg, Value: []byte(evoting.ContractName)},
+		{Key: evoting.FormArg, Value: data},
+		{Key: evoting.CmdArg, Value: []byte(evoting.CmdCombineShares)},
+	}
+
+	_, err = m.addAndWait(args...)
+	if err != nil {
+		return xerrors.Errorf(addAndWaitErr, err)
+	}
+
+	return nil
+}
+
+func encryptManual(message []byte, pubkey kyber.Point) (K, C kyber.Point, remainder []byte, err error) {
+
+	// Embed the message (or as much of it as will fit) into a curve point.
+	M := suite.Point().Embed(message, random.New())
+	max := suite.Point().EmbedLen()
+	if max > len(message) {
+		max = len(message)
+	}
+	remainder = message[max:]
+	// ElGamal-encrypt the point to produce ciphertext (K,C).
+	k := suite.Scalar().Pick(random.New()) // ephemeral private key
+	K = suite.Point().Mul(k, nil)          // ephemeral DH public key
+	S := suite.Point().Mul(k, pubkey)      // ephemeral DH shared secret
+	C = S.Add(S, M)                        // message blinded with secret
+
+	return K, C, remainder, nil
+}
+
+func chunksPerBallot(size int) int { return (size-1)/29 + 1 }
+
+func encodeIDBallot(ID string) types.ID {
+	return types.ID(base64.StdEncoding.EncodeToString([]byte(ID)))
+}
+
+func marshallBallotManual(voteStr string, pubkey kyber.Point, chunks int) (ptypes.CiphervoteJSON, error) {
+
+	ballot := make(ptypes.CiphervoteJSON, chunks)
+	vote := strings.NewReader(voteStr)
+	fmt.Printf("votestr is: %v", voteStr)
+
+	buf := make([]byte, 29)
+
+	for i := 0; i < chunks; i++ {
+		var K, C kyber.Point
+		var err error
+
+		n, err := vote.Read(buf)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to read: %v", err)
+		}
+
+		K, C, _, err = encryptManual(buf[:n], pubkey)
+
+		if err != nil {
+			return ptypes.CiphervoteJSON{}, xerrors.Errorf("failed to encrypt the plaintext: %v", err)
+		}
+
+		kbuff, err := K.MarshalBinary()
+		if err != nil {
+			return ptypes.CiphervoteJSON{}, xerrors.Errorf("failed to marshal K: %v", err)
+		}
+
+		cbuff, err := C.MarshalBinary()
+		if err != nil {
+			return ptypes.CiphervoteJSON{}, xerrors.Errorf("failed to marshal C: %v", err)
+		}
+
+		ballot[i] = ptypes.EGPairJSON{
+			K: kbuff,
+			C: cbuff,
+		}
+	}
+
+	return ballot, nil
+}
+
+// cast vote for the load test
+// return a function to provide a common implementation for
+// the load test and the scenario test
+func castVotesLoad(numVotesPerSec, numSec int) func(BallotSize, chunksPerBallot int, formID, contentType string, proxyArray []string, pubKey kyber.Point, secret kyber.Scalar, t *testing.T) []types.Ballot {
+	return (func(BallotSize, chunksPerBallot int, formID, contentType string, proxyArray []string, pubKey kyber.Point, secret kyber.Scalar, t *testing.T) []types.Ballot {
```

Comment:
This feels really clunky. I didn't take the time to have a deeper look at it, but there must be a better way to do that. Can you think of a better solution ? In worse case we can keep that as "good enough" for now.

Reply:
Yes, that's definitely something that need to be checked but I don't think I will have time to fix it :(

Reply:
I fixed the issue it should be better now